app/src/main/java/com/amos/flyinn/settingsCheck/settingsCheck.java

```diff
@@ -0,0 +1,116 @@
+package com.amos.flyinn.settingsCheck;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.StrictMode;
+import android.provider.Settings;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.amos.flyinn.MainActivity;
+import com.amos.flyinn.summoner.ADBService;
+import com.tananaev.adblib.AdbBase64;
+import com.tananaev.adblib.AdbConnection;
+import com.tananaev.adblib.AdbCrypto;
+
+import java.io.Console;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.HashSet;
+import java.util.List;
+
+import static android.support.v4.content.ContextCompat.startActivity;
+
+public class settingsCheck {
+
+    public enum SettingsType {
+        Developer, ADBperUSB, ADBperNetwork
+    }
+
+    private Context context;
+    private ADBService service = new ADBService();
+    public settingsCheck(Context context) {
+        this.context=context;
+    }
+
+    /**
+     * Collect all disabled settings which are necessary to be enabled
+     * for this app in a list
+     * @return List<Settings> List of disabled settings.
+     */
+    public HashSet<SettingsType> GetMissingSettings() {
+        HashSet<SettingsType> missingSettings=new HashSet<SettingsType>();
+
+        // Developer settings
+        if(android.provider.Settings.Secure.getInt(context.getContentResolver(),
+                android.provider.Settings.Global.DEVELOPMENT_SETTINGS_ENABLED , 0) != 1) {
+            missingSettings.add(SettingsType.Developer);
+        }
+
+        // ADBperUSB settings
+        if(android.provider.Settings.Secure.getInt(context.getContentResolver(),
+                android.provider.Settings.Global.ADB_ENABLED , 0) != 1) {
+            missingSettings.add(SettingsType.ADBperUSB);
+        }
+
+        // ADBperNetwork settings
+        Thread t = new Thread(new Runnable() { // Use own thread because of ThreadPolicy
+            @Override
+            public void run() {
+                StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
```

Comment:
It would be great if this could be changed to running in a separate thread instead of changing thread policies. Please tell if this too much work.

Reply:
Thanks for your review, I added/changed the requested code. 
Changing the threadPolicy wasn't necessary anymore since the commit before.